packages/perseus/src/types.ts

```diff
@@ -548,7 +523,10 @@ export type WidgetExports<
     /**
      * @deprecated - do not use in new code.
      */
-    getUserInputFromSerializedState?: (props: any) => TUserInput;
+    getUserInputFromSerializedState?: (
+        props: any,
+        widgetOptions?: any, // <= for groups
```

Comment:
SerializedState has widget IDs but not widget types. We don't want to determine widget type from widget IDs, but we need the widget type to get the correct `getUserInputFromSerializedState` function for the widget. In ServerItemRenderer/Renderer we already have access to the WidgetOptions but Groups didn't.